### PR TITLE
Adds a deprecation warning about fuel.ignitionrobotics.org

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

## Summary

Outputs a deprecation warning about `fuel.ignitionrobotics.org` when encountered. This will help people transition to `fuel.gazebosim.org`.

Don't merge this until @nkoenig  has updated the models and worlds owned by openrobotics on Fuel, and posted a guide on how others can change their models and worlds.
 
## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/FuelClient.cc [hunk 1]
```diff
@@ -1776,3 +1789,18 @@ void FuelClientPrivate::PopulateLicenses(const ServerConfig &_server)
...
+//////////////////////////////////////////////////
+void FuelClientPrivate::CheckForDeprecatedUri(const common::URI &_uri)
+{
+  static std::string oldServer = "fuel.ignitionrobotics.org";
+  auto ignFuelPos = _uri.Str().find(oldServer);
+  if (ignFuelPos != std::string::npos)
+  {
+    std::string newUrl = _uri.Str();
+    newUrl.replace(ignFuelPos, oldServer.size(), "fuel.gazebosim.org");
+    ignwarn << "The " << oldServer << " URL is deprecrated. Please change "
```
[the remark this answers is not in the document]
Comment (nkoenig) [reply src/FuelClient.cc:1802]: Yes, I already have a script that will work for  anyone. I need to test it, post it, and give people some time to use it before merging this PR.

**Comment by azeey** [comment]
Models that contain `fuel.ignitionrobotics.org` are broken as of #293 for people that build from source. Any way we expedite this?

File: src/FuelClient.cc [hunk 2]
```diff
@@ -186,6 +186,12 @@ class gz::fuel_tools::FuelClientPrivate
   /// license information.
   public: void PopulateLicenses(const ServerConfig &_server);
 
+  /// \brief Checks the provided URI for fuel.ignitionrobotics.org, and
+  /// prints a deprecation warning message if found.
+  /// \param[in] _uri URI to check
+  /// DEPRECATED/DEPRECATION: remove this function in Gazebo H.
```
Comment (azeey) [inline src/FuelClient.cc:192]: I'm not sure removing this for Gazebo H is a good idea. There might still be worlds/models that use `fuel.ignitionrobotics.org` on Fuel well past Gazebo H.

File: src/FuelClient.cc [hunk 3]
```diff
@@ -1776,3 +1789,18 @@ void FuelClientPrivate::PopulateLicenses(const ServerConfig &_server)
     ignerr << "Failed to parse license information[" << resp.data << "]\n";
   }
 }
+
+//////////////////////////////////////////////////
+void FuelClientPrivate::CheckForDeprecatedUri(const common::URI &_uri)
+{
+  static std::string oldServer = "fuel.ignitionrobotics.org";
```
Comment (azeey) [inline src/FuelClient.cc:1796]: We should avoid using types with non-trivial destructors in objects of static storage. Can this be replaced with `static const 
char *oldServer = "fuel.ignitionrobotics.org"`?

File: src/FuelClient.cc (hunk 2 above)
Comment (nkoenig) [reply src/FuelClient.cc]: [765bf6c](https://github.com/gazebosim/gz-fuel-tools/pull/294/commits/765bf6c3b3fbb8cb0e996fbe54951883aa9837b4)

File: src/FuelClient.cc (hunk 3 above)
Comment (nkoenig) [reply src/FuelClient.cc]: [765bf6c](https://github.com/gazebosim/gz-fuel-tools/pull/294/commits/765bf6c3b3fbb8cb0e996fbe54951883aa9837b4)

**Comment by azeey** [comment]
Looks like most of the changes in this PR were included in #304.

**Comment by azeey** [comment]
Closing as a duplicate of #304

**Pull request closed without merging**